cabal-install/src/Distribution/Client/Setup.hs

```diff
@@ -1946,6 +1947,11 @@ installOptions showOrParseArgs =
           installRunTests (\v flags -> flags { installRunTests = v })
           trueArg
 
+      , option [] ["semaphore"]
+          "Use a semaphore so GHC can compile components in parallel"
+          installUseSemaphore (\v flags -> flags { installUseSemaphore = v })
+          trueArg
```

Comment:
maybe this should be a `boolOpt`, so it can be enabled **or disabled** in the global or project config

Reply:
I implemented it now so that if you use an older compiler and pass `--sempahore` then you get a warning about falling back to old behaviour but things still work. That way it can be enabled globally without too much hassle.